src/main/groovy/com/matyrobbrt/gml/mappings/MappingsProvider.groovy

```diff
@@ -16,40 +16,55 @@ import net.minecraftforge.fml.loading.FMLLoader
 import org.apache.commons.codec.binary.Hex
 import org.slf4j.Logger
 
+import java.net.http.HttpClient
+import java.net.http.HttpRequest
+import java.net.http.HttpResponse
 import java.nio.file.Files
 import java.nio.file.Path
 import java.nio.file.StandardCopyOption
 import java.security.MessageDigest
 import java.security.NoSuchAlgorithmException
 import java.util.concurrent.CompletableFuture
+import java.util.concurrent.atomic.AtomicLong
+import java.util.stream.Collectors
+import java.util.zip.GZIPInputStream
 import java.util.zip.ZipEntry
 import java.util.zip.ZipFile
 
 @CompileStatic
 class MappingsProvider {
 
-    private static final Logger LOGGER = LogUtils.getLogger();
-
     public static final MappingsProvider INSTANCE = new MappingsProvider()
 
+    private static final Logger LOGGER = LogUtils.getLogger()
     private static final String PISTON_META = 'https://piston-meta.mojang.com/mc/game/version_manifest_v2.json'
     private static final String MAPPINGS_THREAD = 'GML Mappings Thread'
     private static final String VERSION_JSON = 'version.json'
     private static final String OFFICIAL = 'official.txt'
     private static final String MCPCONFIG_ZIP = 'srg.zip'
     private static final String JOINED_PATH = 'config/joined.tsrg'
-    private static final Gson GSON = new GsonBuilder().setLenient().excludeFieldsWithoutExposeAnnotation().create()
 
-    final String version = FMLLoader.versionInfo().mcVersion()
-    final Path gameDir = Launcher.INSTANCE.environment().getProperty(IEnvironment.Keys.GAMEDIR.get()).get()
-    final Path cacheDir = gameDir.resolve('mod_data/gml')
-    final String mcpVersion = FMLLoader.versionInfo().mcpVersion()
-    final String mcpConfigURL = "https://maven.minecraftforge.net/de/oceanlabs/mcp/mcp_config/${version}-${mcpVersion}/mcp_config-${version}-${mcpVersion}.zip"
+    @Lazy
+    private static final Gson GSON = new GsonBuilder().excludeFieldsWithoutExposeAnnotation().create()
```

Comment:
Making this field lazy will actually bring more overhead, as gson instances are light.

Reply:
In this specific case it involves instantiating at least 4 `ArrayList`s and a `HashMap`. For this reason, along with the fact that GSON is never used after mappings have been setup for the first time, I deemed the `@Lazy` appropriate here.